### PR TITLE
grid catalog

### PR DESCRIPTION
**Pull request by abcdesign1**
- [DEMO LINK](https://vadym4che.github.io/layout_catalog_grid/)

### CODE REVIEW

File: src/styles/_catalog.scss [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+    grid-template-columns: 200px 200px;
+  }
+  @media (min-width: 768px) and (max-width: 1023px) {
+    grid-template-columns: repeat(3, 200px);
+  }
+  @media (min-width: 1024px) {
+    grid-template-columns: repeat(4, 200px);
+  }
 
   &__card {
```
Comment (sergii-nosachenko) [inline src/styles/_catalog.scss:20]: Add width to variables

File: src/styles/_catalog.scss [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+  align-content: center;
+  gap: 48px;
 
-  max-width: 944px;
-  margin: 0 auto;
+  @media (max-width: 487px) {
+    grid-template-columns: 200px;
+  }
+  @media (min-width: 488px) and (max-width: 767px) {
+    grid-template-columns: 200px 200px;
```
Comment (sergii-nosachenko) [inline src/styles/_catalog.scss:11]: Use repeat

File: src/styles/_catalog.scss [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
+  justify-content: center;
+  align-content: center;
+  gap: 48px;
 
-  max-width: 944px;
-  margin: 0 auto;
+  @media (max-width: 487px) {
+    grid-template-columns: 200px;
+  }
+  @media (min-width: 488px) and (max-width: 767px) {
```
Comment (sergii-nosachenko) [inline src/styles/_catalog.scss:10]: You can keep only min-width options. Don't need to combine it with max-width.

File: src/styles/_catalog.scss [hunk 4]
```diff
@@ -1,9 +1,21 @@
...
+    grid-template-columns: 200px 200px;
+  }
+  @media (min-width: 768px) and (max-width: 1023px) {
+    grid-template-columns: repeat(3, 200px);
+  }
+  @media (min-width: 1024px) {
+    grid-template-columns: repeat(4, 200px);
+  }
```
Comment (sergii-nosachenko) [inline src/styles/_catalog.scss:18]: You can make a loop to create all media queries. Just create a correctly mapped list to iterate.

File: src/styles/_catalog.scss [hunk 5]
```diff
@@ -1,9 +1,21 @@
...
+  display: grid;
+  justify-content: center;
+  align-content: center;
+  gap: 48px;
 
-  max-width: 944px;
-  margin: 0 auto;
+  @media (max-width: 487px) {
+    grid-template-columns: 200px;
+  }
```
Comment (sergii-nosachenko) [inline src/styles/_catalog.scss:9]: No need to create a query for the default state.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/styles/_catalog.scss (hunk 2 above)
Comment (vadym4che) [reply src/styles/_catalog.scss:11]: image

File: src/styles/_catalog.scss (hunk 3 above)
Comment (vadym4che) [reply src/styles/_catalog.scss:10]: maybe yes but
if I'll do the way you suggesting me
i'll have a strong need to watch out that next resolution is going exactly next in cascading style sheet file

do you really think I actually have to do all as short as it possible with risk of broking styles scenario?

File: src/styles/_catalog.scss (hunk 4 above)
Comment (vadym4che) [reply src/styles/_catalog.scss:18]: It only will be shortened with previously suggested avoiding max-width I think

File: src/styles/_catalog.scss (hunk 5 above)
Comment (vadym4che) [reply src/styles/_catalog.scss:9]: I'll make complete investigation according to this point and surely give a feedback about this point

File: src/styles/_catalog.scss (hunk 1 above)
Comment (vadym4che) [reply src/styles/_catalog.scss:20]: separated all with  $breakpoints collection

File: src/styles/_catalog.scss (hunk 2 above)
Comment (vadym4che) [reply src/styles/_catalog.scss:11]: @each scss loop applied

File: src/styles/_catalog.scss (hunk 3 above)
Comment (vadym4che) [reply src/styles/_catalog.scss:10]: max-width conditional statements had been removed because of useless

File: src/styles/_catalog.scss (hunk 4 above)
Comment (vadym4che) [reply src/styles/_catalog.scss:18]: got it - have used it - following the best practices

File: src/styles/_catalog.scss (hunk 5 above)
Comment (vadym4che) [reply src/styles/_catalog.scss:9]: unnecessary query states has been removed

**Review by illiarerun — APPROVED** [review]

**Pull request closed without merging**